serving/deploying-with-private-registry.md

```diff
@@ -0,0 +1,211 @@
+# Deploying to Knative using a private container registry
+This guide walks you through deploying an application to Knative from source code in a git repository using a private container registry for the container image. The source code should contain a dockerfile. For this guide, we'll use this [helloworld app](https://github.com/knative/docs/tree/master/serving/samples/helloworld-go), but you could use your own.
+
+
+## Set up a private container registry and obtain credentials
+If you do not want your container image to be publicly available, you may want to use a private container registry. In this example, we'll use IBM Container Registry, but most of these concepts will be similar for other clouds.
+
+1. Ensure you have the [IBM Cloud CLI](https://cloud.ibm.com/docs/cli/reference/ibmcloud/download_cli.html#install_use) installed.  
+
+1. Install the container registry plugin:
+
+    ```
+    ibmcloud plugin install container-registry
+    ```
+
+1. Choose a name for your first namespace, and then create it: 
+
+    ```
+    ibmcloud cr namespace-add <my_namespace>
+    ```
+    
+    A namespace represents the spot within a registry that holds your images. You can set up multiple namespaces as well as control access to your namespaces by using IAM policies.
+
+1. Create a token:
+
+    ```
+    ibmcloud cr token-add --description "token description" --non-expiring --readwrite
+    ```
+    
+    The automated build processes you'll be setting up will use this token to access your images.
+
+1. The CLI output should include a token identifier and the token. Make note of the token. You can verify that the token was created by listing all tokens:
+
+    ```
+    ibmcloud cr token-list
+    ```
+
+## Provide container registry credentials to Knative
+You will use the credentials you obtained in the previous section to authenticate to your private container registry. First, you'll need to create a secret to store the credentials for this registry. This secret will be used to push the built image to the container registry.
+
+A Secret is a Kubernetes object containing sensitive data such as a password, a token, or a key. You can also read more about [Secrets](https://kubernetes.io/docs/concepts/configuration/secret/).
+
+1. Create a file named registry-push-secret.yaml containing the following .yaml.
+
+    ```
```

Comment:
Please add `yaml` to the end so as to highlight the syntax for YAML files, same for other YAML content in this file.

Example like this:

```yaml
    apiVersion: v1
    kind: Secret
    metadata:
      name: registry-push-secret
      annotations:
        build.knative.dev/docker-0: https://registry.ng.bluemix.net
    type: kubernetes.io/basic-auth
    stringData:
      username: token
      password: <token_value>
```

Reply:
Created a suggested change to do this.